### PR TITLE
fix: package types

### PR DESCRIPTION
**Pull request by userquin**
imagen

imagen

### CODE REVIEW

File: scripts/postbuild.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  'utf-8',
+)
+const mDtsModule = resolve('dist/types.d.mts')
+const mDtsModuleContent = readFileSync(mDtsModule, 'utf-8')
+writeFileSync(
+  mDtsModule,
+  mDtsModuleContent.replaceAll(
+    'from \'./module\'',
+    'from \'./module.js\'',
+  ),
```
Comment (danielroe) [inline scripts/postbuild.ts:34]: I think we can safely make these changes in .d.mts files in module builder (but not .d.ts - did you definitely need the latter?)

File: scripts/postbuild.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { resolve } from 'node:path'
+
+// fix cjs export default
+const cjsModule = resolve('dist/module.cjs')
+const cjsModuleContent = readFileSync(cjsModule, 'utf-8')
+writeFileSync(
+  cjsModule,
+  cjsModuleContent.replace(
+    'module.exports = function',
+    'exports.default = function',
```
Comment (danielroe) [inline scripts/postbuild.ts:11]: What did this address?

Comment (userquin) [reply scripts/postbuild.ts:11]: Node10

Comment (userquin) [reply scripts/postbuild.ts:11]: https://arethetypeswrong.github.io/?p=%40vite-pwa%2Fnuxt%400.3.2

File: scripts/postbuild.ts (hunk 1 above)
Comment (userquin) [reply scripts/postbuild.ts:24]: there is no need to patch `types.d.ts`

File: package.json [hunk 3]
```diff
@@ -30,14 +30,14 @@
...
-  "main": "./dist/module.cjs",
+  "main": "./dist/module.mjs",
   "types": "./dist/types.d.ts",
   "files": [
     "dist",
     "*.d.ts"
   ],
   "scripts": {
-    "prepack": "nuxt-module-build prepare && nuxt-module-build build",
+    "prepack": "nuxt-module-build prepare && nuxt-module-build build && esno scripts/postbuild.ts",
```
[the remark this answers is not in the document]
Comment (userquin) [reply package.json]: thx, removed esno

**Comment by userquin** [comment]
I'm going to test changes here with latest Volar (1.8.24) in VSCode and IntelliJ UA and EAP (Beta3)

File: scripts/postbuild.ts (hunk 2 above)
Comment (userquin) [reply scripts/postbuild.ts]: removed from postbuild script

**Review by danielroe — APPROVED** [review]
Looks good. But I think we can patch in module builder directly (just for d.mts) if you want.

**Comment by userquin** [comment]
PR sent to nuxt module builder, once released new version I'll update this PR

**Comment by userquin** [comment]
I'll release a new patch, when nuxt module builder ready we can review this repo

**Pull request merged**